### ISSUE TITLE
Multiple deform armatures/skeletons on same mesh

### ISSUE THREAD

**Issue by Astra307** [playcanvas/engine#5133]
Currently PlayCanvas does not support having multiple deform armatures/skeletons on the same mesh (blending the effect of the different armatures). It seems that it chooses the first armature to be the working one, and the second just doesn’t manage to deform the mesh at all. Though perhaps not very common, support for multiple deform armatures would be very useful in some cases.

Project illustrating this can be found [here](https://playcanvas.com/editor/scene/1683083).
It has 4 different variations of an icosphere that is bound to 2 different deform armatures. This works very well in Blender, where the effect of the 2 different armatures are blended. However in PlayCanvas only the effect of one of the armatures are playing, despite different setups.

Reference [forum post](https://forum.playcanvas.com/t/multiple-deform-armatures-on-same-mesh/30055)

**Comment by mvaligursky** [comment]
I'm not familiar with Blender and its armatures. Are multiple armatures equivalent to multiple skeletons, or multiple animations? Could you please explain / add screenshots of what multiple armatures you have set up.

Here's a guide that might help with how to work / export with this in Blender: https://www.zuluonezero.net/2021/11/16/exporting-multiple-animations-from-blender-to-unity/

**Comment by kungfooman** [comment]
Can you share the original blend file?

The `Bugg-multipleArmatures.glb` has errors:

![image](https://user-images.githubusercontent.com/5236548/223676331-dfcc10ff-c160-4ff7-b70c-4e4200a3ca36.png)

And importing glb back into Blender looks wrong aswell:

![image](https://user-images.githubusercontent.com/5236548/223676930-dcd82628-149e-4d9d-a1a6-40fbe161f3f5.png)

**Comment by Astra307** [comment]
ah, yes. By "multiple armatures" I mean multiple skeletons, sry. 

Here's two Blender files where two armatures/skeletons are used on the same mesh:
In the first one the bones from the two skeletons share the same vertex groups / targets: [file](https://houseofmathnorge-my.sharepoint.com/:u:/g/personal/astrid_houseofmath_no/EdngY91vFChMooJ_UVQczmcBUHOqO6Qfj3s01VpCarIRvQ?e=jcOi36)
In the second one the bones of the different skeletons have entirely different vertex groups / targets: [file](https://houseofmathnorge-my.sharepoint.com/:u:/g/personal/astrid_houseofmath_no/EcNG33FiJnBMiDKuDZJ-MGsBcjLwngXcAwKyi2RpoiZAlg?e=Dlw5f8) 

Basically, the effect from each armature/skeleton separately would be:
![image](https://user-images.githubusercontent.com/29603509/223870983-6a0c313b-7a7b-4197-bee5-322cb02bbfda.png)
![image](https://user-images.githubusercontent.com/29603509/223871527-b393fd69-07d3-4809-89a3-eb10cf9d5492.png)


Combined making a result like this:
![image](https://user-images.githubusercontent.com/29603509/223870712-567e40b0-c8ba-4c59-9e7a-dfa67b7bf041.png)

**Comment by mvaligursky** [comment]
Thanks for the explanation. I believe this should work already, as long as the data are in the correct format. glTF file format supports this. But I assume both skeletons have the animations exported separately, for their sets of bones? In that case, in PlayCanvas, I think two anim components need to be set up on the hierarchy, each playing its set of animations. But I assume its possible to export a single animation for both skeletons, as they're part of the same hierarchy, and a single animation could drive both.
So I would say that is is likely more of a problem of getting data in the correct format / setting up anim components correctly, than the engine issue as such.

**Comment by yaustar** [comment]
I wonder if animation layers would work here so that two animations can be played and blended together

**Comment by Astra307** [comment]
I tried exporting as a single baked animation but couldn't get it to work.. Didn't try the animation layers tho. But I'm pretty sure there's something weird happening with the 2nd armature/sceleton. At least if you try moving one of the bones in the 1st sceleton manually in PC you can see the mesh deform (both in editor and runtime), however moving bones of the 2nd sceleton has no visible effect.

**Comment by mvaligursky** [comment]
I'll close this, as it's not likely we'll have time to investigate this specific issue, as it's likely related to the data set up in the glb file / animations. In general I believe the engine supports this, as this is only two animation playing on separate bones, that are used to skin a single mesh.
